internal/core/attribute.hpp

```diff
@@ -150,6 +150,10 @@ class AttributeMap {
     struct AttributeComputeParam{
         InputIt begin;
         InputIt end;
+        /* size could be exchanged with std::distance, but only as long as
+         * InputIt come from std::vector, so complexity is constant. Passed here
```

Comment:
> but only as long as InputIt come from std::vector

The requirement is not that it's a vector< T >::iterator, It's that the iterator itself is a random access iterator. Which is true for many types of iterators (std::array for example). Or even our own strides one can be made a random access iterator. Essentially the iterator needs to implement `operator += ` to qualify